documentcloud.php

```diff
@@ -25,73 +25,104 @@
     Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
-// Die if this isn't being called from within WordPress.
-defined( 'ABSPATH' ) or die();
-
 class WP_DocumentCloud {
 
+	// Plugin constants
     const CACHING_ENABLED          = true,
           DEFAULT_EMBED_FULL_WIDTH = 940,
           OEMBED_RESOURCE_DOMAIN   = 'www.documentcloud.org',
           OEMBED_PROVIDER          = 'https://www.documentcloud.org/api/oembed.{format}',
           DOCUMENT_PATTERN         = '^(?P<protocol>https?):\/\/www\.documentcloud\.org\/documents\/(?P<document_slug>[0-9]+-[a-z0-9-]+)';
-    
+	/**
+	 * Constructor.
+	 */
     function __construct() {
+		// Check for conflicts with other Documentcloud plugins.
+		// Not needed on WordPress VIP since no other Documentcloud plugins exist.
+		if ( ! defined( 'WPCOM_IS_VIP_ENV' ) || ! WPCOM_IS_VIP_ENV ) {
+			add_action( 'admin_init', array( $this, 'check_dc_plugin_conflict') );
+	        add_action( 'init', array( $this, 'register_dc_oembed_provider' ) );
```

Comment:
@bcampeau Sensible to only `check_dc_plugin_conflict` outside of WP VIP, but `register_dc_oembed_provider` always needs to run, so should be outside of this conditional block. Also, the indentation seems a bit off here?